documentation/modules/post/solaris/escalate/srsexec_readline.md

```diff
@@ -0,0 +1,61 @@
+## Vulnerable Application
+
+  The vulnerable binary (`srsexec`) is included in the package Sun Remote Services
+  Net Connect Software Proxy Core (SUNWsrspx).  The vulnerable versions are
+  3.2.3 and 3.2.4, for Solaris 10.  This package was included on the extras/companion CD which
+  doesn't seem to available anymore.  `srsexec`'s vulnerability is that it runs with the suid bit set,
+  and in debug verbose mode if given a file as input, the first line of that file will be echoed back.
+  Common exploitation is to read `/etc/shadow`, to get root's password hash, however any file can be read.
+
+  In lieu of this, a mock application was created in python and is available
+  [here](https://github.com/h00die/MSF-Testing-Scripts/blob/master/srsexec).
+  Follow the instructions in the python script to install, `argparse` also needs to be sent
+  to the Solaris box since pypi.org doesn't accept ssl2/3 which are the only versions in Solaris 10u9.
+
+  The output from `srsexec` and this module is slightly odd.  The first line of the file will be
+  after `binaries file line: ` and cut at 20 characters.  If the output is longer than 20 characters,
+  then the next line will be the next 20 characters with the first 2 from the previous line repeated.
```

Comment:
This description `will be the next 20 characters with the first 2 from the previous line repeated` is a little confusing. Consider:

```
The output from `srsexec` is slightly odd. The first line of the file will be after `binaries file line: ` and truncated at 20 characters.

If the output is longer than 20 characters, then the next line will start with the last 2 characters from the previous line, followed by the next 18 characters, and so on.
```